panic when referencing pointer on binary type

The following program `sample.go` triggers a panic:

```go
package main

import (
    "fmt"
    "time"
)

func main() {
    v := (*time.Time)(nil)
    fmt.Println(v)
}
```

Expected result:
```console
$ go run ./sample.go
<nil>
```

Got:
```console
$ yaegi ./sample.go
scope.go:151: nil type
panic: nil type

goroutine 1 [running]:
log.Panic(0xc0001af360, 0x1, 0x1)
	/usr/local/Cellar/go/1.12.4/libexec/src/log/log.go:333 +0xac
github.com/containous/yaegi/interp.(*Scope).add(0xc0000a8440, 0x0, 0x11335fa)
	/Users/marc/go/src/github.com/containous/yaegi/interp/scope.go:151 +0x297
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func2(0xc0001dcff0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1086 +0x3dc4
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dcff0, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:104 +0xaf
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dcf00, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dce10, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dcc30, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dcb40, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dc690, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dc000, 0xc0001afd50, 0xc0001afd88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:101 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc00016e120, 0xc0001dc000, 0xc0000a6080, 0x4, 0x0, 0x0, 0x4)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:24 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00016e120, 0xc0001d4000, 0x61, 0x261, 0xc0001d4000, 0x61, 0x0, 0xc0001b1040)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:237 +0x135
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:59 +0x3ec
```

--- a/_test/type8.go
+++ b/_test/type8.go
@@ -1,0 +1,14 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func main() {
+	v := (*time.Time)(nil)
+	fmt.Println(v)
+}
+
+// Output:
+// <nil>

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1075,10 +1075,10 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 			case n.anc.kind == Field:
 				// pointer type expression in a field expression (arg or struct field)
 				n.gen = nop
-			case n.anc.kind == ParenExpr && n.child[0].sym != nil && n.child[0].sym.kind == Typ:
+			case n.child[0].isType(scope):
 				// pointer type expression
 				n.gen = nop
-				n.typ = &Type{cat: PtrT, val: n.child[0].sym.typ}
+				n.typ = &Type{cat: PtrT, val: n.child[0].typ}
 			default:
 				// dereference expression
 				wireChild(n)

--- a/interp/run.go
+++ b/interp/run.go
@@ -171,14 +171,24 @@ func typeAssert2(n *Node) {
 
 func convert(n *Node) {
 	i := n.findex
-	var value func(*Frame) reflect.Value
-	if n.child[1].typ.cat == FuncT {
-		value = genNodeWrapper(n.child[1])
-	} else {
-		value = genValue(n.child[1])
-	}
+	c := n.child[1]
 	typ := n.child[0].typ.TypeOf()
 	next := getExec(n.tnext)
+
+	if c.kind == BasicLit && c.val == nil { // convert nil to type
+		n.exec = func(f *Frame) Builtin {
+			f.data[i] = reflect.New(typ).Elem()
+			return next
+		}
+		return
+	}
+
+	var value func(*Frame) reflect.Value
+	if c.typ.cat == FuncT {
+		value = genNodeWrapper(c)
+	} else {
+		value = genValue(c)
+	}
 
 	n.exec = func(f *Frame) Builtin {
 		f.data[i] = value(f).Convert(typ)

